src/history/__tests__/historyActions.test.ts

```diff
@@ -33,50 +35,67 @@ describe('history actions', () => {
     priceFilter
   })
 
-  beforeEach(() => {
-    store = generateMockStore(mockGlobalState, mockApi)
-    store.clearActions()
-  })
+  const expectedFailedAction = {
+    type: HISTORY_FETCH_FAILED
+  }
 
-  it('should create action update history data', () => {
-    expect(historyReceivedAction(mockHistoryState)).toEqual(expectedHistoryReceivedAction)
-  })
+  describe('fetch success', () => {
+    beforeEach(() => {
+      store = generateMockStore(mockGlobalState, mockApi)
+      store.clearActions()
+    })
 
-  it('should call the mock API correctly', async () => {
-    await store.dispatch(getHistoryData('AAPL') as any)
-    expect(mockApi.getHistory).toHaveBeenCalledWith('AAPL', 'ytd', 'close')
-  })
+    it('should create action update history data', () => {
+      expect(historyReceivedAction(mockHistoryState)).toEqual(expectedHistoryReceivedAction)
+    })
 
-  it('should create HISTORY_RECEIVED_ACTION after successfully fetching history', async () => {
-    const expectedAction = [expectedHistoryReceivedAction]
+    it('should create HISTORY_RECEIVED_ACTION after successfully fetching history', async () => {
```

Comment:
Similarly have the async action in the same 'describe' method. I assume it makes things clear and separate concerns?